.github/workflows/build-docker.yml

```diff
@@ -22,32 +24,50 @@ jobs:
         target: [ release, test ]
     runs-on: ubuntu-latest-8-cores
     permissions:
-      contents: write    
+      contents: write # to update the GH release w/ Docker tags
+      packages: write # to upload images to ghcr
+      id-token: write
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac #v4
         with:
           fetch-depth: 0
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 #v3
-      - name: DockerHub Login
-        uses: docker/login-action@v3
+
+      - name: DockerHub Registry Login
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d
```

Comment:
Just curious, why the hash instead of the named version?

Reply:
tags can be overwritten by repository owners, commits are fixed in git history, hence eliminating the slim change that tag v2 becomes a different commit under the hood